README.md

```diff
@@ -1,46 +1,137 @@
 # sslang
 
-A language built atop the Sparse Synchronous Model
+A language built atop the Sparse Synchronous Model.
 
-This depends on the Haskell Stack and the SSM library
+## Setup
 
-See <https://docs.haskellstack.org/en/stable/GUIDE/> for Haskell Stack
-documentation
+### Requirements
 
-Build the compiler (sslc) with
+We assume that development will take place in a UNIX-like environment (i.e., macOS, WSL, or some kind of Linux distro). Development in Windows is probably possible but unsupported.
+
+### Setup Development Toolchain
+
+You can install this compiler's build toolchain using [GHCup][ghcup], Haskell's toolchain manager, with the following command:
+
+```
+curl --proto '=https' --tlsv1.2 -sSf https://get-ghcup.haskell.org | \
+  BOOTSTRAP_HASKELL_INSTALL_STACK=1 \
+  sh
+```
+
+This will run a short but interactive script; make sure to let it know where it should add the `PATH` variable. It will also ask if you would like to install [HLS][hls] (Haskell Language Server); you should do so if you would like to set up IDE features with an LSP-compatible text editor.
+
+The GHCup setup script may also detect that dependencies are missing, and ask you to install them; make sure to do so before proceeding. For instance, if you are running Ubuntu 20.10:
```

Comment:
Stack's install script is secretly doing this: https://github.com/commercialhaskell/stack/blob/master/etc/scripts/get-stack.sh#L168

GHCup ultimately puts the onus on the user to install system dependencies (though it provides helpful suggestions) because it doesn't want to be in the business of scripting around each system's package manager.